tests/Doctrine/Tests/ORM/Functional/EnumTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Tests\Models\Enums\Card;
+use Doctrine\Tests\Models\Enums\Suit;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use ValueError;
+
+/**
+ * @requires PHP 8.1
+ */
+class EnumTest extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->isSecondLevelCacheEnabled) {
+            $this->markTestSkipped();
+        }
+
+        $this->setUpEntitySchema([
+            Card::class,
+        ]);
+    }
+
+    public function testEnumMapping(): void
+    {
+        $card       = new Card();
+        $card->suit = Suit::Clubs;
+
+        $this->_em->persist($card);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $fetchedCard = $this->_em->find(Card::class, $card->id);
+
+        $this->assertInstanceOf(Suit::class, $fetchedCard->suit);
+        $this->assertEquals(Suit::Clubs, $fetchedCard->suit);
+    }
+
+    public function testEnumWithNonMatchingDatabaseValueThrowsException(): void
+    {
+        $card       = new Card();
+        $card->suit = Suit::Clubs;
+
+        $this->_em->persist($card);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $metadata = $this->_em->getClassMetadata(Card::class);
+        $this->_em->getConnection()->update(
+            $metadata->table['name'],
+            [$metadata->fieldMappings['suit']['columnName'] => 'invalid'],
+            [$metadata->fieldMappings['id']['columnName'] => $card->id]
+        );
+
+        $this->expectException(ValueError::class);
```

Comment:
I think we should catch and upcast this error. There is something wrong with the user's database and the user should be made aware of that. DBAL has a `ConversionException` for errors like this one. Either we reuse that or we declare our own exception for that.

Reply:
I think we should have our own one, it would be less confusing.